docs/design/resource_controller.md

```diff
@@ -0,0 +1,269 @@
+# Kubernetes Proposal: ResourceController
+
+**Related PR:** 
+
+| Topic | Link |
+| ----- | ---- |
+| Admission Control Proposal | https://github.com/GoogleCloudPlatform/kubernetes/pull/2501 |
+| Separate validation from RESTStorage | https://github.com/GoogleCloudPlatform/kubernetes/issues/2977 |
+
+## Background
+
+This document proposes a system for enforcing resource limits as part of admission control.
+
+## Model Changes
+
+A new resource, **ResourceController**, is introduced to enumerate resource usage constraints scoped to a Kubernetes namespace.
+
+Authorized users are able to set the **ResourceController.Spec** fields to enumerate desired constraints.
+
+```
+// ResourceController is an enumerated set of resources constraints enforced as part of admission control
+type ResourceController struct {
+  TypeMeta   `json:",inline"`
+  ObjectMeta `json:"metadata,omitempty"`
+  // Spec represents the imposed constraints for allowed resources
+  Spec ResourceControllerSpec `json:"spec,omitempty"`
+  // Status represents the observed allocated resources to inform constraints
+  Status ResourceControllerStatus `json:"status,omitempty"`
+}
+
+type ResourceControllerGroupBy string
+type ResourceControllerRuleType string
+
+const (
+  ResourceControllerGroupByPod                   ResourceControllerGroupBy = "Pod"
+  ResourceControllerGroupByNamespace             ResourceControllerGroupBy = "Namespace"
+  ResourceControllerGroupByContainer             ResourceControllerGroupBy = "Container"
+  ResourceControllerGroupByReplicationController ResourceControllerGroupBy = "ReplicationController"
+
+  ResourceControllerRuleTypeMin ResourceControllerRuleType = "Min"
+  ResourceControllerRuleTypeMax ResourceControllerRuleType = "Max"
+)
+
+// ResourceControllerGroup is a named grouping of a rule and resources
+type ResourceControllerGroup struct {
+  GroupBy   ResourceControllerGroupBy  `json:"groupBy,omitempty"`
```

Comment:
Or did you mean that you could only set the same Allocated for all buckets of that group-by?